### PR TITLE
feat(ADVISOR-2955): updating react router to v6 using new fec components

### PR DESCRIPTION
**Pull request by Fewwy**
# Description
Update of react router utilizing new components from the frontend components

Associated Jira ticket: # (issue)
https://issues.redhat.com/browse/ADVISOR-2955

Use this google sheet to review my PR https://docs.google.com/spreadsheets/d/1JlOaYrGNRrxgSaHTaJudEwKptw3cEvlsB20fJdDqMqc/edit?usp=sharing

### CODE REVIEW

File: config/dev.webpack.config.js
Comment (bastilian) [inline config/dev.webpack.config.js:31]: If it's not needed for the react router upgrade, let's keep this out.

Comment (Fewwy) [reply config/dev.webpack.config.js:31]: Ok, will fix that :)

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:105]: We need to preserve this empty state.

File: src/Routes.js
Comment (mkholjuraev) [inline src/Routes.js:40]: This import is also needed as this component is rendered on line 89. Also, the component might need to be changed to TopicAdmin as on line 89.

Comment (Fewwy) [reply src/Routes.js:40]: It is present. Check the line 153

File: src/Routes.js
Comment (Fewwy) [reply src/Routes.js]: I added a ZeroState.js file that could be used in the element props for Route.
The only issue I have is the Pathways details throwing a 404 (I'm looking into why it happens), everything else works.

**Comment by bastilian** [comment]
@Fewwy The links in the navigation/sidebar are not working correctly. They should not trigger a page refresh. The breadcrumbs do also not appear to be correctly. And the paths of the routes still have a `/` at the start which they should not, and which is likely the cause for breadcrumbs and/or sidebar links not working.

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:58]: Why not wrap the `Routes` containing all `Route` components in one `ZeroState`?

File: src/Routes.js
Comment (Fewwy) [reply src/Routes.js:40]: Agree, that is a better way to do that

File: src/Routes.js
Comment (Fewwy) [reply src/Routes.js:58]: Agree, was waiting for other pair of eyes to look at this zerostate I wrote to see if it any good

File: src/PresentationalComponents/Export/SystemsPdfBuild.js
Comment (bastilian) [inline src/PresentationalComponents/Export/SystemsPdfBuild.js:87]: Watch out! You are replacing React **PDF** `Link` component with a React Router `Link` (Wrapper component). This one can stay as it is.

File: src/PresentationalComponents/Breadcrumbs/Breadcrumbs.js
Comment (bastilian) [inline src/PresentationalComponents/Breadcrumbs/Breadcrumbs.js:63]: The constructing of the `navigate` in `buildBreadcrumbs` seems to be borked.

**Comment by bastilian** [comment]
Links on the recommendation details like this one(Apache HTTP Server), appear to not work correctly: Screenshot 2023-06-28 at 00 38 29

**Comment by bastilian** [comment]
@Fewwy We might need to make a change to the advisor components[0] in order to work under both v5 and v6. But this should be easy by just using the `InsightsLink` there, which should respect the app & bundle, but let's look into that together.


[0] https://github.com/RedHatInsights/frontend-components/blob/master/packages/advisor-components/src/common.tsx#L22

File: src/PresentationalComponents/Export/SystemsPdfBuild.js
Comment (Fewwy) [reply src/PresentationalComponents/Export/SystemsPdfBuild.js]: Yep, it works after I added Link back

File: src/PresentationalComponents/Breadcrumbs/Breadcrumbs.js
Comment (Fewwy) [reply src/PresentationalComponents/Breadcrumbs/Breadcrumbs.js]: I had to bring the Link back, otherwise it would break when I try to go back to the dashboard using breadcrumbs

**Comment by Fewwy** [comment]
> @Fewwy We might need to make a change to the advisor components[0] in order to work under both v5 and v6. But this should be easy by just using the `InsightsLink` there, which should respect the app & bundle, but let's look into that together.
> 
> [0] https://github.com/RedHatInsights/frontend-components/blob/master/packages/advisor-components/src/common.tsx#L22

Sure, I can do that, I checked the functionality of the Links and they are working fine now, but we can replace them if this is necessary

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:142]: Where did this go?

**Review by AsToNlele — APPROVED** [review]
Looks like everything works, good job @Fewwy! Lets wait for another review

**Review by mkholjuraev — APPROVED** [review]
LGTM, I have a few minor enhancement suggestions. Feel free to address them in a separate PR.

File: src/ZeroState.js
Comment (Fewwy) [reply src/ZeroState.js]: I added that

File: src/ZeroState.js
Comment (Fewwy) [reply src/ZeroState.js]: Thanks, I added that

File: src/PresentationalComponents/RulesTable/helpers.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/helpers.js:498]: Thanks I included that :)

**Pull request merged**
